examples/step-90/step-90.cc

```diff
@@ -0,0 +1,1344 @@
+    const DoFHandler<dim>                  &level_set_dof_handler;
+    const VectorType                       &level_set;
+
+    // In addition to FEValues an unfitted quadrature for the main
+    // solution, we need a cell quadrature
+    //  for the FE space of the level-set.
+    FEValues<dim>              level_set_fe_values;
+    NonMatching::FEValues<dim> non_matching_fe_values;
+  };
+
+
+  // @sect2{Copy objects for TraceFEM}
+  // This CopyData object is customized for TraceFEM.
+  //  In particular, the implementation of the normal-gradient volume
+  //  stabilization relies on it.
+
+  template <int dim>
+  struct TraceFEM_CopyData
+  {
+    FullMatrix<double>                   cell_matrix;
+    Vector<double>                       cell_rhs;
+    std::vector<types::global_dof_index> local_dof_indices;
+    unsigned int                         cell_index;
+    double                               cell_L2_error_sqr;
+    double                               cell_H1_error_sqr;
+    double                               cell_stab_error_sqr;
+
+    template <class Iterator>
+    void reinit(const Iterator &cell, const unsigned int dofs_per_cell)
+    {
+      cell_matrix.reinit(dofs_per_cell, dofs_per_cell);
+      cell_rhs.reinit(dofs_per_cell);
+      local_dof_indices.resize(dofs_per_cell);
+      cell->get_dof_indices(local_dof_indices);
+    }
+
+    template <class Iterator>
+    void reinit(const Iterator &cell)
+    {
+      cell_index          = cell->active_cell_index();
+      cell_L2_error_sqr   = 0;
+      cell_H1_error_sqr   = 0;
+      cell_stab_error_sqr = 0;
+    }
+  };
+
+  // @sect2{Normal-gradient stabilization form of TraceFEM}
+  // The following class corresponds to the stabilization form, its contribution
+  // to the global matrix and to the error.
+  template <int dim>
+  class NormalGradientVolumeStabilization
+  {
+  public:
+    NormalGradientVolumeStabilization(VectorType &solution,
+                                      VectorType &level_set)
+      : solution(solution)
+      , level_set(level_set)
+      , stabilization_parameter(1.0)
+      , stabilization_exponent(-1.0)
+    {}
+
+    // Since  the stabilization form contributes to the error computation,
+    //  we pass the discrete solution and the discrete level-set to the class
+    //  which refer to different dof-handlers.
+    VectorType &solution;
+    VectorType &level_set;
+    // Clearly, the next method indicates that the assembly and the evaluation
+    // of the form does require a bulk cell quadrature. This methodology may be
+    // utilized in the MeshWorker.
+    bool needs_cell_worker()
+    {
+      return true;
+    }
+    // We define the stabilization form here assuming that ScratchData and
+    // CopyData arguments are initialized properly. The local contribution of
+    // the stabilization on this cell is defined.
+    template <class Iterator>
+    void cell_worker_assemble(const Iterator            &cell,
```

Comment:
Let's call this argument `IteratorType` to make clear it's a type. Do you need this to be a template parameter, though, or would `typename DoFHandler<dim>::cell_iterator` (or `...::active_cell_iterator`) be enough?